packages/next/src/client/components/app-router.tsx

```diff
@@ -207,13 +207,13 @@ function useChangeByServerResponse(
 function useNavigate(dispatch: React.Dispatch<ReducerActions>): RouterNavigate {
   return useCallback(
     (href, navigateType, forceOptimisticNavigation, shouldScroll) => {
-      const url = new URL(addBasePath(href), location.href)
+      const url = new URL(addBasePath(href), window.location.href)
```

Comment:
For consistency with other cases.

Reply:
Could `location.*` (and conversely, `history.*` too) be used everywhere? Surely repeating window increases the bundle size (albeit it probably compresses better).